pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1126,31 +1126,75 @@ func (s *TracerSuite) TestSelfConnect() {
 	}, 5*time.Second, 100*time.Millisecond, "could not find expected number of tcp connections, expected: 2")
 }
 
+// sets up two udp sockets talking to each other locally.
+// returns (listener, dialer)
+func setupUdpSockets(udpnet, ip string) (*net.UDPConn, *net.UDPConn, error) {
+	serverAddr := fmt.Sprintf("%s:%d", ip, 0)
+
+	laddr, err := net.ResolveUDPAddr(udpnet, serverAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ln, c *net.UDPConn = nil, nil
+	defer func() {
```

Comment:
`t.Cleanup` would be preferable to `defer` here. You can pass in `t *testing.T` as a parameter to the function.

Reply:
Thanks, updated